code/parse/sexp.cpp

```diff
@@ -2464,8 +2465,8 @@ int check_sexp_syntax(int node, int return_type, int recursive, int *bad_node, i
 				}
 				Assert(ship_node >= 0);
 
-				// we can't check special-arg ships
-				if (Sexp_nodes[ship_node].flags & SNF_SPECIAL_ARG_IN_NODE)
+				// we can't check nodes whose value can change
+				if (can_node_value_change(node))
```

Comment:
I know I should check both here and in similar cases that the node is specifically one of a special argument, string variable, or string container data before skipping the check. I'll add that before I mark the PR as ready for review.